pkg/common/configs/configvalidator.go

```diff
@@ -706,7 +728,7 @@ func checkQueuesStructure(partition *PartitionConfig) error {
 	if rootQueue.Resources.Guaranteed != nil || rootQueue.Resources.Max != nil {
 		return fmt.Errorf("root queue must not have resource limits set")
 	}
-	return checkQueues(&rootQueue, 1)
+	return nil
```

Comment:
pardon, is this change related to this PR?

Reply:
Yes, we need the `checkLimits` inside `checkQueues` to check root queue limits (after partition limits is applied to root queue), so here extracted the method out of `checkQueuesStructure`

Reply:
ok, you want to the root queue gets all fixes (limits and inserts) before calling `checkQueues`.

for another, you change the behavior or `checkQueuesStructure` without changing test-related code. It seems `checkQueuesStructure` has no tests, right? If so, could you file jira to complete units for each `checkXXX`

Reply:
filed https://issues.apache.org/jira/browse/YUNIKORN-2297